### PR TITLE
Git "fixup!" convention

### PR DESCRIPTION
**Pull request by samholmes**

### CODE REVIEW

File: git.md
Comment (swansontec) [inline git.md]: Git has a built-in convention, where you can do "fixup! Old commit message" as the commit message. Then, if you do a `rebase -i` with the correct flag enabled (don't remember which), it will automatically put this as a fixup commit right after the commit with the message "Old commit message".

Comment (swansontec) [reply git.md]: `git config --global rebase.autosquash true`, then you can do `git commit --squash <offending-commit-hash>`, and git will create the proper commit message for this to happen, so it's all automatic. I think we should embrace this practice in this document somehow.

**Review by swansontec — CHANGES_REQUESTED** [review]
Git has some built-in stuff to help with this, which we should use.

File: git.md
Comment (samholmes) [reply git.md]: A quick google into what you're talking about and I found this: https://thoughtbot.com/blog/autosquashing-git-commits This appears to reflect what you have mentioned perfectly. Furthermore, I do like the use of fixup over squash in this case (no need to keep the commit's message). I say you're right on here, and I should repurpose this PR to suggest the proper us of "fixup" commits during the review process. Good call!

**Review by swansontec — APPROVED** [review]
Approved, but it's not clear if the requester or the reviewer is the one doing the final rebase / push / merge.

**Comment by samholmes** [comment]
> Approved, but it's not clear if the requester or the reviewer is the one doing the final rebase / push / merge.

Good point. I'll make it clearer by rephrasing. After thinking it through, I think the convention should be that the requester rebases and merges. The work has been reviewed and approved and all that's left is tidying up, which we should expect the requester to have the wherewithal to complete.

EDIT: On second thought, either the requester or reviewer can merge (though I'd say it's the requester's responsibility to take the initiative).

**Pull request merged**
